### PR TITLE
inventory code contribution for container orchestration

### PR DESCRIPTION
**Pull request by GregoryIvo**
Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to inderstand this section can be skipped.

**Screenshots:** If applicable, add screenshots to help explain your solution

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

File: kura/org.eclipse.kura.core.inventory/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -13,6 +13,7 @@ Import-Package: org.apache.commons.io;version="1.4.9999",
  org.eclipse.kura.cloudconnection.message;version="[1.0,2.0)",
  org.eclipse.kura.cloudconnection.request;version="[1.0,1.1)",
  org.eclipse.kura.configuration;version="[1.1,2.0)",
+ org.eclipse.kura.container.orchestration.provider;version="[1.0.0,2.0.0)";resolution:=optional,
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.inventory/META-INF/MANIFEST.MF]: This is usually [1.0,2.0)

File: kura/org.eclipse.kura.core.inventory/OSGI-INF/inventory.xml [hunk 2]
```diff
@@ -24,5 +24,6 @@
               interface="org.eclipse.kura.cloudconnection.request.RequestHandlerRegistry"/>
    <reference bind="setSystemService" cardinality="1..1" interface="org.eclipse.kura.system.SystemService" name="SystemService" policy="static" unbind="unsetSystemService"/>
    <property name="service.pid" type="String" value="org.eclipse.kura.core.inventory.InventoryHandlerV1"/>
-   <reference bind="setDeploymentAdmin" cardinality="1..1" interface="org.osgi.service.deploymentadmin.DeploymentAdmin" name="DeploymentAdmin" policy="static" unbind="unsetDeploymentAdmin"/>>
+   <reference bind="setDeploymentAdmin" cardinality="1..1" interface="org.osgi.service.deploymentadmin.DeploymentAdmin" name="DeploymentAdmin" policy="static" unbind="unsetDeploymentAdmin"/>
+      <reference cardinality="1..1" interface="org.eclipse.kura.container.orchestration.provider.DockerService" name="DockerService" policy="static" bind="setDockerService" unbind="unsetDockerService"/>
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.inventory/OSGI-INF/inventory.xml]: The reference is set to 1..1 but the import is optional. What will happen if the container service is not available?

File: kura/org.eclipse.kura.core.inventory/src/main/java/org/eclipse/kura/core/inventory/resources/DockerContainer.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eurotech and/or its affiliates and others
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.inventory/src/main/java/org/eclipse/kura/core/inventory/resources/DockerContainer.java]: 2022

File: kura/org.eclipse.kura.core.inventory/src/main/java/org/eclipse/kura/core/inventory/resources/DockerContainers.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eurotech and/or its affiliates and others
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.inventory/src/main/java/org/eclipse/kura/core/inventory/resources/DockerContainers.java]: 2022

File: kura/org.eclipse.kura.json.marshaller.unmarshaller.provider/src/main/java/org/eclipse/kura/internal/json/marshaller/unmarshaller/system/JsonJavaDockerContainersMapper.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eurotech and/or its affiliates and others
```
Comment (MMaiero) [inline kura/org.eclipse.kura.json.marshaller.unmarshaller.provider/src/main/java/org/eclipse/kura/internal/json/marshaller/unmarshaller/system/JsonJavaDockerContainersMapper.java]: 2022

File: kura/org.eclipse.kura.xml.marshaller.unmarshaller.provider/src/main/java/org/eclipse/kura/internal/xml/marshaller/unmarshaller/XmlJavaDockerContainersMapper.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 Eurotech and/or its affiliates and others
```
Comment (MMaiero) [inline kura/org.eclipse.kura.xml.marshaller.unmarshaller.provider/src/main/java/org/eclipse/kura/internal/xml/marshaller/unmarshaller/XmlJavaDockerContainersMapper.java]: New file so it should be just 2022

File: kura/test/org.eclipse.kura.core.inventory.test/META-INF/MANIFEST.MF [hunk 7]
```diff
@@ -7,6 +7,7 @@ Bundle-Vendor: Eclipse Kura
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Fragment-Host: org.eclipse.kura.core.inventory
 Import-Package: org.eclipse.kura;version="[1.2,2.0)",
+ org.eclipse.kura.container.orchestration.provider;version="[1.0.0,2.0.0)",
```
Comment (MMaiero) [inline kura/test/org.eclipse.kura.core.inventory.test/META-INF/MANIFEST.MF]: Usually [1.0,2.0)

File: kura/org.eclipse.kura.core.inventory/OSGI-INF/inventory.xml (hunk 2 above)
Comment (MMaiero) [reply kura/org.eclipse.kura.core.inventory/OSGI-INF/inventory.xml]: This is still open, in my opinion. We should support cases where the container orchestration is not there but the inventory still works.
The dependency should be 0..1 or even potentially 0..n

**Comment by MMaiero** [comment]
Tests are still failing for me:
```
[ERROR] Errors:
[ERROR]   InventoryHandlerV1Test.testStartContainerWithoutVersion:954->givenTheFollowingJsonKuraPayload:1005 » Kura
[ERROR]   InventoryHandlerV1Test.testStopContainerWithoutVersion:964->givenTheFollowingJsonKuraPayload:1005 » Kura
[INFO]
[ERROR] Tests run: 25, Failures: 0, Errors: 2, Skipped: 0
```

File: kura/org.eclipse.kura.core.inventory/OSGI-INF/inventory.xml (hunk 2 above)
Comment (GregoryIvo) [reply kura/org.eclipse.kura.core.inventory/OSGI-INF/inventory.xml]: Changed this to 0..1, as I don't see the possibility of running more than one container engine connector (of the same type) on the same host.  Thoughts?

**Pull request merged**